proto/ord_schema.proto

```diff
@@ -99,78 +123,22 @@ message Compound {
   repeated CompoundIdentifier identifiers = 1;
   // Quantity of this compound used in the reaction.
   Amount amount = 2;
-  // TODO(ccoley): fix this bastardization of proto and json
-  /**
-   * The "components" field is only used when species are not added to the
-   * reaction vessel in their pure form. 
-   *
-   * For pre-mixed solutions, we define all components of what went in
-   * to the stock solution. For example, suppose we are adding
-   * 3 mL of a 4 M solution of NaOH in water.
-   * 
-   * compound {
-   *   identifiers: [
-   *     {type: IDENTIFIER_NAME, value: "4M NaOH solution in water"}
-   *   ]
-   *   amount {
-   *     volume: {value: 3, units: VOLUME_UNITS_MILLILITER}
-   *   }
-   *   components: [
-   *     {
-   *       identifiers: [
-   *         {type: IDENTIFIER_SMILES, value: "O"},
-   *         {type: IDENTIFIER_NAME, value: "water"}
-   *       ]
-   *       amount: {
-   *         volume: {value: 3, units: VOLUME_UNITS_MILLILITER}
-   *       }
-   *     }
-   *     {
-   *       identifiers: [
-   *         {type: IDENTIFIER_SMILES, value: "[Na+].[OH-]"},
-   *         {type: IDENTIFIER_NAME, value: "sodium hydroxide"}
-   *       ]
-   *       amount {
-   *         concentration: {value: 4, units: CONCENTRATION_UNITS_MOLAR}
-   *       }
-   *     }
-   *   ]
-   * }
-   *
-   * Equivalently, if 1 L of the stock solution had been prepared,
-   * 
-   * compound {
-   *   identifiers: [
-   *     {type: IDENTIFIER_NAME, value: "4M NaOH solution in water"}
-   *   ]
-   *   amount {
-   *     volume: {value: 3, units: VOLUME_UNITS_MILLILITER}
-   *   }
-   *   components: [
-   *     {
-   *       identifiers: [
-   *         {type: IDENTIFIER_SMILES, value: "O"},
-   *         {type: IDENTIFIER_NAME, value: "water"}
-   *       ]
-   *       amount: {
-   *         volume: {value: 1, units: VOLUME_UNITS_LITER}
-   *       }
-   *     }
-   *     {
-   *       identifiers: [
-   *         {type: IDENTIFIER_SMILES, value: "[Na+].[OH-]"},
-   *         {type: IDENTIFIER_NAME, value: "sodium hydroxide"}
-   *       ]
-   *       amount {
-   *         moles: {value: 4, units: MOLES_UNIT_MOLES}
-   *       }
-   *     }
-   *   ]
-   * }
-   *
-   * A similar definition is used for supported catalysts (TODO)
-   */
-  repeated Compound components = 3;
+  enum ReactionRole {
```

Comment:
I think I still prefer having this at the level of `ReactionInput`; if the input is limiting, you add more of that input, not more of the component compound...right?

Reply:
I struggled with this decision a bit. The reason why I think ReactionRoles should be assigned to pure species (compounds) is that we should be able to parse one input into its constituent components. I see two potential use cases:
- When templating reactions, if we have an input with one REACTANT component and one SOLVENT component, we can write python wrappers to easily exchange the REACTANT (holding the SOLVENT constant, or vice versa) by iterating through the ```components``` without having to know the order components are defined in.
- We can programmatically generate a reaction SMILES for a reaction proto that does not yet have it defined by looking for all REACTANT compounds. This would let us easily generate reaction SMILES strings that, by default, exclude solvents that are presumed to be spectators. Similarly, if we have a solid supported catalyst with one CATALYST component--palladium--and one UNKNOWN (SUPPORT?) component--carbon--we can determine that the catalytically active component is palladium.

Reply:
SGTM; I think I was confusing this with our conversation about an enum (that doesn't exist) for solute vs. solvent, etc.